yb-voyager/src/query/queryissue/detectors.go

```diff
@@ -554,3 +554,41 @@ func (j *JsonPredicateExprDetector) GetIssues() []QueryIssue {
 	}
 	return issues
 }
+
+type CommonTableExpressionDetector struct {
+	query                      string
+	materializedClauseDetected bool
+}
+
+func NewCommonTableExpressionDetector(query string) *CommonTableExpressionDetector {
+	return &CommonTableExpressionDetector{
+		query: query,
+	}
+}
+
+func (c *CommonTableExpressionDetector) Detect(msg protoreflect.Message) error {
+	if queryparser.GetMsgFullName(msg) != queryparser.PG_QUERY_CTE_NODE {
+		return nil
+	}
+	/*
+	with_clause:{ctes:{common_table_expr:{ctename:"cte" ctematerialized:CTEMaterializeNever 
+	ctequery:{select_stmt:{target_list:{res_target:{val:{column_ref:{fields:{a_star:{}} location:939}} location:939}} from_clause:{range_var:{relname:"a" inh:true relpersistence:"p" location:946}} limit_option:LIMIT_OPTION_DEFAULT op:SETOP_NONE}} location:906}} location:901} op:SETOP_NONE}} stmt_location:898
+	*/
+	cteNode, err := queryparser.ProtoAsCTENode(msg)
+	if err != nil {
+		return err
+	}
+	if cteNode.Ctematerialized != queryparser.CTE_MATERIALIZED_DEFAULT {
```

Comment:
Can you add a comment here explaining that both MATERIALIZED, NOT MATERIALIZED flags were not supported..